CMakeLists.txt

```diff
@@ -67,7 +67,11 @@ function(link_to_source base_name)
         if (CMAKE_HOST_UNIX)
             set(command ln -s ${target} ${link})
         else()
-            set(command cmd.exe /c mklink /j ${link} ${target})
+            if (IS_DIRECTORY ${target})
```

Comment:
The documentation of `mklink` tells me that `/j` “Creates a Directory Junction”, but I have no idea why you'd use one as opposed to the default mode. It does sound like it wouldn't work for non-directories. Why is `/j` used for directories?

Reply:
I'm not clued up on the intricacies of exactly how junctions works, but Windows differentiates between symlinks for files and directories. I'll do a deeper dive into the documentation.

Reply:
I'm no expret, but I believe junctions can only be to folders, but symbolic links can be to either folders or files. On some systems, you may be able to create a junction where you couldn't create any symbolic links due to permissions issues (fewer permissions required for junction making), but on our test system we seem to have permission enough to make symlinks.

Since we can make symlinks, was using the `/d` option to `mklink` considered? It creates a directory symbolic link.

Reply:
It's not just for our test systems. #688 changed from `/d` to `/j` for permission reasons.

Reply:
This may introduce a regression, then. Since it goes back to using symbolic links for files, the permissions elevation may be required again.

Since the `sh` files aren't needed by Windows anyway, could we avoid making links for them and get away with only making junctions?

Reply:
Some of the `.sh` files _are_ needed by Windows. They're how the Windows tests are invoked. I don't remember whether we currently run SSL tests in a CMake build on Windows on our CI, but this should work.

If we can't make symbolic links work on Windows, then we should copy the files. This has the downside that the build directory will contain stale copies if the original file has been edited, so the build scripts should be modified to make the a build product of the original that gets rebuilt (copied) when the original is updated, preferably only on Windows.

Reply:
I did suggest when we reviewed the original PR #1470, that it should also be able to copy files, and raised it as issue #1496.

I still think that work needs to be done, but I think we can defer it for now, and just fix the use of  `mklink` as I suggest elsewhere.